sql/src/main/java/com/vmware/ddlog/util/sql/CalciteDDLVisitorBase.java

```diff
@@ -0,0 +1,78 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.vmware.ddlog.util.sql;
+
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlNodeList;
+import org.apache.calcite.sql.ddl.SqlCreateTable;
+import org.apache.calcite.sql.util.SqlBasicVisitor;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * To translate DDL statements in Calcite SQL, the Calcite visitor should visit each node of the DDL statement.
+ * However, any DDL statements that we are interested in are of type SqlNodeList, so this part can be shared
+ * across any Calcite-to-\<dialect\> translator.
+ *
+ * Any subclass of CalciteDDLVisitorBase should override visit(SqlCall call) and add the result to translatedColumns.
+ */
+public class CalciteDDLVisitorBase extends SqlBasicVisitor<String> {
+    // Holds the return values from visit(SqlCall call);
+    protected List<String> translatedColumns = new ArrayList();
+
+    @Override
+    public String visit(SqlNodeList nodeList) {
+        for (SqlNode statement : nodeList) {
+            if (statement instanceof SqlCreateTable) {
+                SqlCreateTable createStatement = (SqlCreateTable) statement;
+
+                // For the SqlCreateTable class, getOperandList returns [name, columnList, query]
+                List<SqlNode> operands = createStatement.getOperandList();
+
+                if (operands.get(1) instanceof SqlNodeList) {
+                    SqlNodeList columns = (SqlNodeList) operands.get(1);
+                    for (SqlNode column : columns) {
+                        if (column instanceof SqlCall) {
+                            // The return value is ignored; we assume the overridden method will add
+                            // the result to translatedColumns
+                            visit((SqlCall) column);
+                        } else {
+                            throw new UnsupportedOperationException("Cannot visit SqlNodes that aren't SqlCall type");
```

Comment:
I would say "cannot translate". "visit" is very technical and will confuse end-users of the system who don't know how the implementation works.

Reply:
Exceptions that are uncaught are effectively error messages for end users of the system.